fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1549,10 +1571,27 @@ private Bundle processEntriesForMethod(Bundle requestBundle, Bundle responseBund
         log.entering(this.getClass().getName(), "processEntriesForMethod", new Object[] {"httpMethod", httpMethod });
         
         try {
-            // Visit each of the request entries, processing those with the specified request method.
+            // First, obtain a list of request entry indices for the entries that we'll process.
+            // This list will contain the indices associated with only the entries for the specified http method.
+            List<Integer> entryIndices =
+                    getBundleRequestIndicesForMethod(requestBundle, responseBundle, httpMethod);
+            if (log.isLoggable(Level.FINER)) {
+                log.finer("Bundle request indices to be processed: " + entryIndices.toString());
+            }
+
+            // Next, for PUT and DELETE requests, we need to sort the indices by the request url path value.
+            if (httpMethod.equals(HTTPVerb.PUT) || httpMethod.equals(HTTPVerb.DELETE)) {
+                sortBundleRequestEntries(requestBundle, entryIndices);
```

Comment:
I think @punktilious requested to add this one back in "to avoid deadlock issues", but I'm hoping someone can explain it to me because I'm not sure I get it.

Reply:
here is the problematic scenario:

within single transaction bundle:
```
bundle 1:
PUT A
PUT B
```

within a separate transaction bundle:
```
bundle 2:
PUT B
PUT A
```

the timing problem would be:
1. bundle 1 arrives first, it goes through entries and when it hits PUT A it locks A
2. bundle 2 arrives very soon afterward, and it starts processing the PUT B before bundle 1's PUT B (locking B)
3. deadlock